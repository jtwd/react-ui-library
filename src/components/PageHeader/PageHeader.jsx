import React from 'react'
import { node } from 'prop-types'

import { Header, Title } from './PageHeader.styles'

function PageHeader({title, children}) {
  return (
    <Header>
      <Title>{title}</Title>
      {children}
    </Header>
  )
}

PageHeader.propTypes = {
  /** Title content */
  title: node.isRequired,
  /** Any extra content - eg. meta data etc */
  children: node
}

PageHeader.defaultProps = {
  children: null
}

export default PageHeader