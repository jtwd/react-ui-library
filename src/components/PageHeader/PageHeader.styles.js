import styled from 'styled-components'

import { themeProps } from '../_theme/utils'
import theme from './PageHeader.theme'

const t = themeProps(theme)

export const Header = styled.header`
  margin: ${t.headerMargin};
  border-bottom: ${t.borderBottom};
`

export const Title = styled.h1`
  font-size: ${t.fontSize};
  margin: ${t.h1Margin};
  small {
    font-size: ${t.smallFontSize};
    color: ${t.smallColor};
    display: inline-block;
    padding: ${t.smallPadding};
  }
`
