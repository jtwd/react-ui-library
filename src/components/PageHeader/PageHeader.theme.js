import { getTheme } from '../_theme/utils'
import * as c from '../_theme/palette'

const { spacing, fontSizes } = getTheme()

export default {
  default: {
    headerMargin: `0 0 ${spacing.xl}`,
    borderBottom: `1px solid ${c.trans(false)}`,
    fontSize: fontSizes.huge,
    h1Margin: `0 0 ${spacing.md}`,
    color: c.light(),
    smallFontSize: '75%',
    smallColor: c.trans(false),
    smallPadding: `0 0 0 ${spacing.sm}`
  }
}
