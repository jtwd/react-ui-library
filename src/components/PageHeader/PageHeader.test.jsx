import React from 'react'
import renderer from 'react-test-renderer'
import PageHeader from './PageHeader'

test('PageHeader renders correctly when passed a title string', () => {
  const component = renderer.create(<PageHeader title="Page title" />)
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('PageHeader renders correctly when passed a title component', () => {
  const Title = <span>Page title <small>with sub title</small></span>
  const component = renderer.create(<PageHeader title={Title} />)
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('PageHeader renders correctly when passed a title component and extra content', () => {
  const component = renderer.create(<PageHeader title="Page title"><p>Extra content</p></PageHeader>)
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
