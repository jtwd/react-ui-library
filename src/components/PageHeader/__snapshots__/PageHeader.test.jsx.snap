// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageHeader renders correctly when passed a title component 1`] = `
<header
  className="sc-bdVaJa fxervv"
>
  <h1
    className="sc-bwzfXH icSllj"
  >
    <span>
      Page title 
      <small>
        with sub title
      </small>
    </span>
  </h1>
</header>
`;

exports[`PageHeader renders correctly when passed a title component and extra content 1`] = `
<header
  className="sc-bdVaJa fxervv"
>
  <h1
    className="sc-bwzfXH icSllj"
  >
    Page title
  </h1>
  <p>
    Extra content
  </p>
</header>
`;

exports[`PageHeader renders correctly when passed a title string 1`] = `
<header
  className="sc-bdVaJa fxervv"
>
  <h1
    className="sc-bwzfXH icSllj"
  >
    Page title
  </h1>
</header>
`;
